File: python/connect4.py
import game
# Main Program


def print_board(board_field):
    i = 0
    j = 0
    while i < 6:
        print(repr(i + 1) + ": ", end='', flush=True)
        while j < 7:
            if board_field[i][j] is None:
                print(" ", end='', flush=True)
            else:
                if board_field[i][j] == 0:
                    print("o", end='', flush=True)
                else:
                    print("x", end='', flush=True)
            j += 1
        print("")
        i += 1
        j = 0
    print("   1234567")

game = game.Game()
print_board(game.board.field)
while True:
    input_var = int(input('Next Move: '))
    input_var -= 1
    game.place(input_var)
    game.generate_computer_decision()
    print_board(game.board.field)
    if game.status != 0:
        break
    print(game.iterations)
print("Done.")

File: cpp/Connect4.cpp
#include "stdafx.h"
#include "Game.h"
#include "Board.h"


void printBoard(vector<vector<int8_t>> board_field);

int main()
{
	Game game;
	game.place(3);
	printf("Test1");
	game.generateComputerDecision();
	printf("Test2");
	getchar();
	return 0;
}

void printBoard(vector<vector<int8_t>> board_field)
{
	for (int row = 0; row < 6; row++)
	{
		printf("%d: ", (row + 1));
		for (int column = 0; column < 7; column++)
		{
			if (board_field[row][column] == 0)
				printf(" ");
			else if (board_field[row][column] == 1)
				printf("o");
			else if (board_field[row][column] == 2)
				printf("x");
			else
				printf("Error on Row: %d and Column: %d \n", row, column);
		}
		printf("\n");
	}
	printf("   1234567");
}